### ISSUE TITLE
$pageimage->maxSize() crops image

### ISSUE THREAD

**Issue by Toutouwai** [processwire/processwire-issues#57]
### Short description of the issue

$pageimage->maxSize() can crop images when it should always preserve the image proportions.

### Expected behavior

The [blog post](https://processwire.com/blog/posts/processwire-2.6.8-brings-new-version-of-reno-admin-theme-and-more/#new-pageimage-gt-maxsize-width-height-method) that introduces maxSize() says:
> $thumb = $image->maxSize($width, $height); 
> This will return an image no larger than the given dimensions, without cropping or anything that would modify the proportions. Of course, this also means that it's likely one of the dimensions (width or height) will be less than what you asked for, since nothing can be cropped.

### Actual behavior

I have an image that is 1200x600 and if I do $my_image->maxSize(400, 400)->url I get an image that is cropped 400x400 when it should be uncropped 400x200.

### Optional: Suggestion for a possible fix

In the forums Zeka suggested:
```
public function maxSize($width, $height, $options = array()) {
	$w = $this->width();
	$h = $this->height();
	if($w >= $h) {
		return $this->maxWidth($width, $options);
	} else {
		return $this->maxHeight($height, $options);
	}
}
```

Alternatively maxSize() could be an alias for size() with 'cropping' set to false in the $options array. 

### Steps to reproduce the issue

1. Upload an image sized 1200 x 600 pixels.
2. Call `$pageimage->maxSize(400, 400)` on the image.

### Setup/Environment

* ProcessWire version: 3.0.37
* (Optional) PHP version: 7

**Comment by ryancramerdesign** [comment]
Thanks @Toutouwai I've pushed the fix for this. I like these fixes that result in less code, I wonder what I was thinking with the extra code that was in there. There must have been some reason for it, but I can't find any practical purpose so see no reason to keep it.

**Comment by BernhardBaumrock** [comment]
hi @ryancramerdesign 
this issue is still not fixed at version 3.0.63! see my screenshots of tests in the forum: https://processwire.com/talk/topic/14567-pageimage-maxsize-not-working-as-described/?do=findComment&comment=145949

it's an easy fix:
```
	public function maxSize($width, $height, $options = array()) {
		$defaults = array(
			'upscaling' => false,
			'cropping' => false
			);
		$options = array_merge($defaults, $options);
		return $this->size($width, $height, $options);
	}
```
